File: src/swift-camera/IPPGSignalsCollector.swift
import Foundation

// Swift equivalent of IPPGSignalsCollector.cs
protocol IPPGSignalsCollector {
    var signalsCount: Int { get }
    var qualityWarnings: Int { get }
    var data: [PpgSignal] { get }
    
    func processFrame(_ frame: Frame)
}

File: src/original-c#-camera/IPPGSignalsCollector.cs
using System.Collections.Generic;

namespace RivaDigitalApp
{
    public interface IPPGSignalsCollector
    {
        int SignalsCount { get; }

        int QualityWarnings { get; }

        void ProcessFrame(Frame frame);

        IEnumerable<PpgSignal> Data { get; }
    }
}
